plugins/osdn/podinfoserver.go

```diff
@@ -0,0 +1,138 @@
+package osdn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/gorilla/mux"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+)
+
+type PodInfoServer struct {
+	http.Server
+
+	registry *Registry
+	vnidMap  *VNIDMap
+	nodeName string
+
+	clusterNetwork string
+	nodeNetwork    string
+	nodeGateway    string
+	mtu            uint
+}
+
+func NewPodInfoServer(r *Registry, v *VNIDMap, nodeName string) *PodInfoServer {
+	router := mux.NewRouter()
+ 
+	s := &PodInfoServer{
+		registry: r,
+		vnidMap:  v,
+		nodeName: nodeName,
+		Server:   http.Server{
+			Handler: router,
+		},
+	}
+
+	router.NotFoundHandler = http.HandlerFunc(http.NotFound)
+	router.Methods("GET").Path("/network").HandlerFunc(s.getNetwork)
```

Comment:
Talked this over with Jordan and David.  I think from a security perspective I can live with an API and a unix domain socket.  Some things that need to change / happen in order to merge it:

1. The handlers and structs sent back and forth need to be more strictly versioned and handled.  I.e., the path needs to be prefixed with a version `/v1/` and the objects should be in a versioned API package `sdn/localserver/api/v1` package and well documented
2. I want to nuke the watches from the SDN server component - it should use a kube client that is passed in (by the openshift node process) to make calls to the server, and potentially add an LRU cache.
3. I would recommend factoring your plugin as an http.Handler that is going to be instantiated by the node code (not directly used from inside openshift-sdn) for now
4. Use the existing Kube client instead of inventing your own - that handles rate limiting, etc
5. We'll have to sort through the unix domain socket and how it interacts with packaging, but at least for step 1 we are still going to run openshift node with both kubelet and SDN plugin (turned on) - the node process will boot the server, and the expectation the SDN sub binary will run inside of the node container.